pkg/common/util/spiffeid.go

```diff
@@ -0,0 +1,111 @@
+package util
```

Comment:
See relevant comment here: https://github.com/spiffe/spire/pull/422#discussion_r180833176

Perhaps `validation` or `idutil` or something like that?

Re-thinking the validation methods will make this a lot more useful... I can think of the following cases:
* SPIFFE ID is well-formed (any trust domain, any path including none)
* SPIFFE ID is well-formed and belongs to our trust domain
* SPIFFE ID is well-formed, belongs to our trust domain, and has a non-zero path (i.e. it is a valid workload id)

UPDATE: I see this has been merged already. Leaving the comment anyways for posterity

Reply:
Oh, very relevant and seem like a quick addition. I'll put together a PR real quick.

Thanks!